schema/encoding/jsonschema/benchmark_test.go

```diff
@@ -38,15 +39,25 @@ func BenchmarkEncoder(b *testing.B) {
 			},
 		},
 		{
-			Name:   `Schema=Student`,
+			Name:   `Schema=Simple`,
 			Schema: studentSchema,
 		},
+		{
+			Name:   `Schema=Complex1`,
+			Schema: requestSchema,
+		},
+		{
+			Name:   `Schema=Complex2`,
+			Schema: responseSchema,
+		},
 	}
+	buf := bytes.NewBuffer(make([]byte, 65535))
```

Comment:
On reflection, I actually think the Discard idea is probably not a good idea because the writes are a part of the performance of the API. The memory bandwidth of the test host would affect the outcome and if we use Discard we wouldn't get a measurement of that. The only thing these changes do is hoist of the allocation Buffer itself outside of the measurement loop.